library/alloc/src/sync.rs

```diff
@@ -4119,11 +4119,11 @@ unsafe fn data_offset<T: ?Sized>(ptr: *const T) -> usize {
     // and extern types, the input safety requirement is currently enough to
     // satisfy the requirements of align_of_val_raw; this is an implementation
     // detail of the language that must not be relied upon outside of std.
-    unsafe { data_offset_align(align_of_val_raw(ptr)) }
+    unsafe { data_offset_align(Alignment::new_unchecked(align_of_val_raw(ptr))) }
```

Comment:
Oh, no `Alignment::of_val` or `Alignment::of_val_raw`.  Huh, guess this is best then.